src/analyses/goto_check.cpp

```diff
@@ -1747,6 +1749,26 @@ void goto_checkt::check_rec(const exprt &expr, guardt &guard)
     if(check_rec_member(to_member_expr(expr), guard))
       return;
   }
+  else if(expr.id() == ID_forall || expr.id() == ID_exists)
+  {
+    // In the case of a quantified statement, skolemise the expression
+    // and recurse.
+    const auto quantified_expr = to_quantifier_expr(expr);
+    const auto quantifier_sym = quantified_expr.symbol();
+    const auto new_symbol = get_fresh_aux_symbol(
+      quantifier_sym.type(),
+      CPROVER_PREFIX,
+      id2string(ns.lookup(quantifier_sym.get_identifier()).base_name),
+      quantified_expr.source_location(),
+      ID_C,
```

Comment:
Given that this code is not part of the C front end, hard coding the language to C seems like the wrong choice. I suggest using the language mode of `quantifier_sym`.

Reply:
This is still hard coded in your new `rename_and_substitute` function and still needs fixing.